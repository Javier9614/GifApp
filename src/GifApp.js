import React, { useState } from "react";
import { AddCategories, GifGrid } from './components';


const GifApp = ({defaultCategories = []}) => {
    const [categories, setCategories]=useState(defaultCategories);

    // const handleAdd = () => {
    //     setCategories([...categories,"HunterXHunter"]);
    // }

    return(
        <>
            <h2>GifApp</h2>
            <AddCategories setCategories={ setCategories }/>
            <hr />
            
            <ul>
                {
                categories.map(category => {
                    return <GifGrid 
                    key={category}
                    category={category}
                    />;        
                })
                }
            </ul>
        </>
    )
}

export default GifApp;