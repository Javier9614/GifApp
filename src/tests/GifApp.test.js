import React from 'react';
import { shallow } from "enzyme";
import GifApp from "../GifApp";

describe("Pruebas en <GifApp>",()=>{

    test("Comprobamos que <GifApp> se muestra correctamente",()=>{

        const wrapper = shallow(<GifApp/>);
        expect(wrapper).toMatchSnapshot();
    })

    test("Debe mostrar una lista de categorias",()=>{

        const  categories = ["Konosuba", "Bleach"];
        const wrapper = shallow(<GifApp defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("GifGrid").length).toBe(categories.length);

    })

})