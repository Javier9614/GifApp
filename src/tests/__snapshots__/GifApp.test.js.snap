// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifApp> Comprobamos que <GifApp> se muestra correctamente 1`] = `
<Fragment>
  <h2>
    GifApp
  </h2>
  <AddCategories
    setCategories={[Function]}
  />
  <hr />
  <ul />
</Fragment>
`;

exports[`Pruebas en <GifApp> Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifApp
  </h2>
  <AddCategories
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      category="Konosuba"
      key="Konosuba"
    />
    <GifGrid
      category="Bleach"
      key="Bleach"
    />
  </ul>
</Fragment>
`;
